src/composables/mutations/useSignOutMutation.js

```diff
@@ -0,0 +1,45 @@
+import { useMutation, useQueryClient } from '@tanstack/vue-query';
+import { useRouter } from 'vue-router';
+import * as Sentry from '@sentry/vue';
+import { useAuthStore } from '@/store/auth';
+import { SIGN_OUT_MUTATION_KEY } from '@/constants/mutationKeys';
+import { APP_ROUTES } from '@/constants/routes';
+
+/**
+ * Sign-Out mutation.
+ *
+ * @returns {Object} The mutation object returned by `useMutation`.
+ */
+const useSignOutMutation = () => {
+  const authStore = useAuthStore();
+  const router = useRouter();
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationKey: SIGN_OUT_MUTATION_KEY,
+    mutationFn: async () => {
+      await authStore.roarfirekit.signOut();
+    },
+    onSuccess: async () => {
+      // Cancel all actively fetching queries.
+      await queryClient.cancelQueries();
+
+      // Reset store and delete persisted data. Persisted data should be cleared via the $reset but to be safe, we also
+      // remove it manually from sessionStorage to prevent any issues.
+      authStore.$reset();
+      sessionStorage.removeItem('authStore');
+      sessionStorage.removeItem('gameStore');
+
+      // Redirect to sign-in page.
+      await router.push({ path: APP_ROUTES.SIGN_IN });
+
+      // Clear the query client to remove all cached data.
+      queryClient.clear();
```

Comment:
Not a change request but just a question: does ordering matter here? Is there a difference between routing to the sign-in page then clearing the query client and clearing the client before routing to sign in?

If not, would it be better to clear before routing so that we further guarantee that we arrive at sign-in with no cached data?

Reply:
Excellent question! In this case, order does usually matter. 

If we were to execute `queryClient.clear()` before routing, the user should see data flash/disappear in front of their eyes for a brief moment before being redirected to the sign-in page. That's why I usually opt to `await` the routing result and only remove cached data from the query client then.  

However, I tested this again and it doesn't really be a concern here – especially since we're already resetting auth store state before the redirect. So I moved the cache removal to happen before the routing.

Should we observe some ugly UX during sign-out, I'll happily go back and improve it.